appendices.tex

```diff
@@ -370,6 +370,149 @@ \chapter{Listing of application roles}
+% 	reason
+% 	\item[DBI\_TRACE] @@@TODO
+% 	\item[pathsep] @@@TODO
+% 	\item[latex] Identifies if the system has \LaTeX available: 0 means no \LaTeX, 1 means \LaTeX supported
+% 	\item[check\_max\_invoices] @@@TODO
+% 	\item[max\_post\_size] States the maximum size in bytes of the body in case of a POST request; this
+% 	setting reduces the impact of DoS requests, if the system is under attack. The setting also limits the
+% 	maximum size of attachments (see \secref{sec-workflows-quotations-file-attachments}) that can be uploaded
+% 	\item[decimal\_places] @@@TODO
+% 	\item[cookie\_name] @@@TODO
+% 	\item[no\_db\_str] @@@TODO
+	
+\end{description}
+
+\subsubsection{[environment] Section}
+\label{subsubsec-global-config-ledgersmb-conf-environment}
+
+	Sets shell environment variables when starting up.
+
+\begin{description}[style=nextline]
+	\item[PATH] If the server can't find applications, append to the path. For example:
+	\texttt{PATH=/bin:/usr/bin:/usr/local/bin:/usr/local/pgsql/bin}
+	\item [PERL5LIB] On some operating systems like macOS the perl path may need to be set. For example, on macOS using Fink's Perl libs it might be set like:
+	\texttt{PERL5LIB=/sw/lib/perl5}
+\end{description}
+
+
+\subsubsection{[paths] Section}
+\label{subsubsec-global-config-ledgersmb-conf-paths}
+
+\begin{description}[style=nextline]
+	\item [templates] Templates base directory. Typically set to \texttt{templates}.
+	\item [images] Images base directory. Typically set to \texttt{UI/images}.
+	\item [localepath] Path to locale files. Defaults to \texttt{locale/po}.
+	\item [templates\_cache] Location where compiled templates are stored. When relative, appended to the directory specified by \texttt{File::Spec->tmpdir()}. 
+	
+	Defaults to \texttt{lsmb\_templates}.
+	\item [workflows] Directory where workflow files are stored. Defaults to \texttt{workflows}
+	\item [custom\_workflows] Directory where custom workflow files are stored. Custom workflows are used to override behavior of the default workflows by providing actions, conditions, etc. with the same name and type or by providing workflows of the same type with additional states and actions. Defaults to \texttt{custom\_workflows}.
+	
+% Items from the old book
+%	\item [spool] @@@TODO
+%	\item [userspath] @@@TODO
+%	\item [memberfile] @@@TODO
+\end{description}
+
+\subsubsection{[programs] Section}
+\label{subsubsec-global-config-ledgersmb-conf-programs}
+
+Setup the various helper programs that can be used with LedgerSMB.
+
+% From the old book.
+%\begin{description}[style=nextline]
+%	\item [gzip] @@@TODO
+%\end{description}
+
+
+\subsubsection{[mail] Section}
+\label{subsubsec-global-config-ledgersmb-conf-mail}
+
+\begin{description}[style=nextline]
+	\item [backup\_email\_from] Set to enable e-mail delivery of backups. For example:
+	\texttt{backup\_email\_from = backups@lsmb\_hosting.com}
+
+% From the old Book
+%	\item [sendmail] @@@TODO
+%	\item [smtphost] @@@TODO

```

Comment:
Still applies: https://github.com/ledgersmb/LedgerSMB/blob/master/old/lib/LedgerSMB/Sysconfig.pm#L89 ; smtphost takes precedence over sendmail. It's the IP address or host name of the SMTP host to be used to inject mail.

Reply:
In looking at https://github.com/ledgersmb/LedgerSMB/blob/9b1b3279c4e7673a1b70bfb762f96b6fed87da2d/old/lib/LedgerSMB/Sysconfig.pm#L119-L127

Is that code correct at line 124? You really select 'tls' to get 'ssl'?

Reply:
For now, update to the way it looks like the code behaves.